pkg/host/store.go

```diff
@@ -86,6 +86,7 @@ func (s *Store[E]) Create(ctx context.Context, obj E) (E, error) {
 	}
 
 	obj.SetCreatedAt(time.Now())
+	obj.SetGeneration(obj.GetGeneration() + 1)
```

Comment:
i looked at official kubernetes api concept and they have written this:

```
generation
integer	A sequence number representing a specific generation of the desired state. Populated by the system. Read-only.
```

Desired state is for us spec structure.
Second problem what i see in future is extend of status with ObservedGeneration logic. we don't have ObservedGeneration in status currently but if we add it, we can end in infinity loop how ObservedGeneration will want to achieve Generation.

 So i prefer here don't use generation but extend metedata with resourceVersion and use it.
@so-sahu, @hardikdr, @afritzler what do you think?

Reply:
Your suggestion to avoid using the generation field and instead extend metadata with resourceVersion aligns with Kubernetes API concepts and is a valid approach in this scenario. 

Concurrency control and consistency in Kubernetes are achieved through several mechanisms, including the use of resourceVersion. More details on how Kubernetes handles concurrency control and consistency can be found in its API conventions document: [Concurrency Control and Consistency in Kubernetes API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency).

Reply:
Done